### PR TITLE
Migrate CertificatesSigningRequests API to V1

### PR DESCRIPTION
**Pull request by palexster**
# Description

This PR migrates CertificateSigningRequest API to v1 from v1beta1. V1beta1 is deprecated and will be removed in a future version of Kubernetes.
This is related to #627 which upgraded K8s libraries to v1.21.

### CODE REVIEW

**Review by aleoli — APPROVED** [review]

File: pkg/vkMachinery/csr/doc.go [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Package csr contains the logic required to generate, create and approve CSR for the Virtual Kubelet instances.
```
Comment (giorio94) [inline pkg/vkMachinery/csr/doc.go:1]:
```suggestion
// Package csr contains the logic required to generate, create and approve Certificate Signing Requests (CSR) for the Virtual Kubelet instances.
```

File: cmd/init-virtual-kubelet/main.go [hunk 2]
```diff
@@ -42,42 +47,24 @@ func main() {
...
+	var crtChan = make(chan []byte)
+	var cert []byte
+	informer := csr.ForgeInformer(cancel, client, name, crtChan)
+	go informer.Run(cancelCtx.Done())
+	select {
+	case _ = <-cancelCtx.Done():
+		klog.Fatalf("Unable to get certificate: timeout elpased")
+	case cert = <-crtChan:
+		if err := utils.WriteFile(vk.CertLocation, cert); err != nil {
+			os.Exit(1)
```
Comment (giorio94) [inline cmd/init-virtual-kubelet/main.go:64]: I suppose this should deserve an error message

File: pkg/vkMachinery/csr/approve.go [hunk 3]
```diff
@@ -39,5 +70,48 @@ func WatchCSR(clientset k8s.Interface, label string, resyncPeriod time.Duration)
...
+	var fieldSelector = map[string]string{
+		"metadata.name": name,
+	}
+	lo := func(options *metav1.ListOptions) {
+		options.FieldSelector = labels.SelectorFromSet(fieldSelector).String()
+	}
+	options := []informers.SharedInformerOption{
+		informers.WithTweakListOptions(lo),
+	}
+	informer := informers.NewSharedInformerFactoryWithOptions(client, resyncPeriod, options...)
```
Comment (giorio94) [inline pkg/vkMachinery/csr/approve.go:88]:
```suggestion
	informer := informers.NewSharedInformerFactoryWithOptions(client, resyncPeriod, informers.WithTweakListOptions( func(options *metav1.ListOptions) {
		options.FieldSelector = labels.SelectorFromSet(fieldSelector).String()
	}))
```

Should be equivalent and less verbose

**Review by giorio94 — APPROVED** [review]

**Pull request merged**